examples/CMakeLists.txt

```diff
@@ -19,6 +19,9 @@ pkg_check_modules(JxlThreads REQUIRED IMPORTED_TARGET libjxl_threads)
 add_executable(decode_oneshot decode_oneshot.cc)
 target_link_libraries(decode_oneshot PkgConfig::Jxl PkgConfig::JxlThreads)
 
+add_executable(decode_progressive decode_progressive.cc)
+target_link_libraries(decode_progressive PkgConfig::Jxl PkgConfig::Jxl lodepng)
```

Comment:
"lodepng" is not defined in this context. This is an example project that a user may have. To compile this example with lodepng as a libjxl user we would need to also include a way to depend on lodepng from the example project. lodepng in particular is not a good project to depend on because it is not available in many places, and there's also no support on the package for easily include the libraries/headers paths (like a pkg-config .pc or a .cmake file would provide). This complexity outweighs any nice to have of an example that writes .png.

Reply:
I see. Maybe write .pam instead (can be done without dependencies, doesn't store icc though) and leave the png writing code there but commented out?

Reply:
Modified it to write .pam. The code to write png with lodepng is in a comment, for those who want to do that instead, but this way there is no dependency besides libjxl itself.